// turkcell/main.js
'use strict'
const crypto = require('crypto');
const qr = require('qr-image');
const fs = require('fs');

function encode(str, key) {
	var pword = key||'aes-256-ctr';
	var encoder = crypto.createCipher('aes-256-ctr', pword);
	return encoder.update(str, 'utf8','hex');
}

function decode(str, key) {
	var pword = key||'aes-256-ctr';
	var decoder = crypto.createDecipher('aes-256-ctr', pword);
	return decoder.update(str, 'hex','utf8');
}

function qrgen(data, file){
    var dataToEncode = data || null;
    var outImage = file || null;
    if(dataToEncode !== null && outImage !== null) {
        qr.image(dataToEncode, {
            type : 'png',
            size :20
        }).pipe(fs.createWriteStream(outImage));
        return true;
    } else{
        return false;
    }
}

function sum(a, b) {
    return a + b;
}

function minus(a, b) {
    return a - b;
}

function selamla() {
	console.log("Herkese selam, hoş geldiniz.");
}

function sakir(isim) {
	if(!isim)
		console.log("Herkese benden çay, şakire yok.");
	else
		console.log("Herkese benden çay,", isim,"e yok :)"); 
}
	
function sarikutu_yeniabone(msisdn) {
	console.log("Yapim asamasindadir.", msisdn);
}

//module.exports = {}
module.exports.encode = encode;
module.exports.decode = decode;
module.exports.qrgen = qrgen;
module.exports.topla = sum;
module.exports.cikar = minus;
module.exports.selamla = selamla;
module.exports.sakir = sakir;
module.exports.sarikutu_yeniabone = sarikutu_yeniabone;
